R/ignore.R

```diff
@@ -22,5 +22,10 @@ use_build_ignore <- function(files, escape = TRUE, base_path = ".") {
 escape_path <- function(x) {
   x <- gsub("\\.", "\\\\.", x)
   x <- gsub("/$", "", x)
+
+  # pattern `<driveletter>:\user\...` causes the PCRE regex compilation error,
```

Comment:
Why? I don't understand what the problem is

Reply:
When normalizePath() is called upstream, given a full path, plus when it goes through escape path, the resulting string will be written to the .Rbuildignore:

`C:\User\hadley\...`

per the discussion https://github.com/hadley/pkgdown/issues/307, The next time the .Rbuildignore is parsed, such as when calling `devtools::build()` it will present with the following error (taken from #pkgdown/307) 

```
Error in grepl(e, files, perl = TRUE, ignore.case = TRUE) : 
  invalid regular expression '^C:\Users\gabriel_odom\Documents\GitHub\mvMonitoring\mvMonitoring\docs$'
Execution halted
```

In order to have a valid regex, we need the first driveletter to have a double slash `C:\\...`

Therefore, line 29 will now add back an extra `\` if the pattern is detected.

Reply:
Why is it only the drive letter that needs the extra escape?

Why isn't line 23 already adding the extra escapes?